### PR TITLE
Keep proxying if resource is not found

### PR DESCRIPTION
**Pull request by XVincentX**
#805 point 3

[Review with no whitespaces](https://github.com/stoplightio/prism/pull/808/files?utf8=✓&diff=unified&w=1)

This PR will short circuit the error handler in case the `errors` flag is set to false and the proxy is on — so we can still proxy the request although all the validation checks will not be performed since we have no idea what resource the user is trying to hit.

### CODE REVIEW

File: packages/core/src/factory.ts [hunk 1]
```diff
@@ -71,25 +72,51 @@ export function factory<Resource, Input, Output, Config extends IPrismConfig>(
...
-        TaskEither.map(({ output, resource, inputValidations }) => {
-          const outputValidations = config.validateResponse
-            ? pipe(
-                Option.fromEither(Either.swap(components.validateOutput({ resource, element: output }))),
-                Option.getOrElse<IPrismDiagnostic[]>(() => [])
-              )
-            : [];
+        TaskEither.fold(
+          error => {
+            if (!config.errors && isProxyConfig(config)) {
```
Comment (StefanDywersant) [inline packages/core/src/factory.ts]: I have a feeling that the code got complicated. I believe adding more separation between proxying and mocking would improve a lot here.

Below is kind of "work in progress" however, it shows clearly the idea of how to simplify the flow:
```
       return pipe(
        isProxyConfig(config)
          ? pipe(
            TaskEither.fromEither(components.route({ resources, input })),
            TaskEither.chain(resource => inputValidation(resource, input, config)),
            TaskEither.alt(error => config.errors ? TaskEither.left(error) : TaskEither.right({ inputValidations: [
                {
                  message:
                    "The selected route hasn't been found and the errors is set false. Prims will proxy the request to the upstream server but no validation will happen",
                  severity: DiagnosticSeverity.Warning,
                },
              ] })),
            TaskEither.chain(({ inputValidations }) => forward(input, config, inputValidations)),
          )
        : pipe(
            TaskEither.fromEither(components.route({ resources, input })),
            TaskEither.chain(resource => inputValidation(resource, input, config)),
            TaskEither.chain(({ resource, inputValidations }) => mock(resource, input, config, inputValidations)),
          ),
        TaskEither.map(outputValidation)
      );
```

This requires separating `forward` and `mock` and ofc some polishing around typing (I'll be happy to do that if you want).

File: packages/http-server/src/__tests__/body-params-validation.spec.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ function instantiatePrism2(operations: IHttpOperation[]) {
       validateRequest: true,
       validateResponse: true,
       mock: { dynamic: false },
+      errors: false,
```
[the remark this answers is not in the document]
Comment (XVincentX) [reply packages/http-server/src/__tests__/body-params-validation.spec.ts]: > I'd lean towards the state when our 4 main components require minimal configuration on their external interfaces.

This is already happening. This is an internal object where all the properties are required; in the external client they're almost all optional and we provide sensible defaults.

File: packages/core/src/factory.ts (hunk 1 above)
Comment (XVincentX) [reply packages/core/src/factory.ts]: Yeah I agree the flow is getting complicated — I'm willing to try your solution. I'll get back on this and see if that will bring some improvements.

Comment (XVincentX) [reply packages/core/src/factory.ts]: Ok I have tried and the resulting flow isn't really that simplified. I would temporarily keep it in this way but if you're willing to give it a spin — be my guest! Just open a PR starting from this brach and we'll go through together!

**Pull request merged**